client/watch.js

```diff
@@ -1036,6 +1069,22 @@ Template.context_browser.helpers({
   mediaTypeForDisplay (){
     return pluralizeMediaType(Session.get('mediaDataType') || Session.get('previousMediaDataType')).toUpperCase();
   },
+  replayAvailable(){
+    console.log(mainPlayer.activeStream.get());
+    
+    if(mainPlayer.activeStream.get().source === "youtube" && !mainPlayer.activeStream.get().live){
```

Comment:
I could be wrong, but when the activeStream goes from live to not-live, or vice-versa, I don't think this will update like you expect it to, since the *reactiveVar* 'mainPlayer.activeStream' isn't getting `set` anywhere when that happens.